### PR TITLE
close menu after confirming avatar

### LINKED ISSUE #9049: [Bug] Selecting an avatar no longer closes the menu on mobile

### PR DESCRIPTION
**Pull request by aditya-mitra**
## Summary

create avatar through readyplayer was working fine

### <samp>🤖 Generated by Copilot at d98dbe7</samp>

Added a popup menu for avatar selection in the user menu. The popup menu is triggered by clicking on the avatar icon and uses the `showPopupMenu` method of the `PopupMenuServices` class.

## References
closes #9049 

### <samp>🤖 Generated by Copilot at d98dbe7</samp>

* Show a popup menu of avatars when the user clicks on the avatar icon in the user menu (`AvatarSelectMenu.tsx`, [link](https://github.com/EtherealEngine/etherealengine/pull/9090/files?diff=unified&w=0#diff-5b9433b387297dc0c6ed55db1406c2fd9fae45fd0b569dbee5c096f0a671ff7cR86))

### <samp>🤖 Generated by Copilot at d98dbe7</samp>

> _Click on avatar_
> _`showPopupMenu` appears_
> _Autumn of choices_

## QA Steps

11f8934e-b140-40c3-b5d4-3ca31dbcd2c6

## Checklist
- If this PR is still a WIP, convert to a draft
- When this PR is ready, mark it as "Ready for review"
- [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- Changes have been manually QA'd 
- Changes reviewed by at least 2 approved reviewers

### CODE REVIEW

**Review by HexaField — CHANGES_REQUESTED** [review]
Rather than just closing the menu, we should add a spinner over the `Confirm` button until the avatar has loaded (can react to `EngineState userReady`)

**Comment by HexaField** [comment]
> Rather than just closing the menu, we should add a spinner over the `Confirm` button until the avatar has loaded (can react to `EngineState userReady`)

https://github.com/EtherealEngine/etherealengine/pull/9097

this PR might be necessary to make this work properly

File: packages/client-core/src/user/components/UserMenu/menus/AvatarSelectMenu.tsx [hunk 1]
```diff
@@ -58,6 +60,8 @@ const AvatarMenu = () => {
   const authState = useHookstate(getMutableState(AuthState))
   const userId = authState.user?.id?.value
   const userAvatarId = useHookstate(getMutableState(AvatarState)[Engine.instance.userID].avatarID as EntityUUID)
+  const avatarLoading = useHookstate(false)
+  const isUserReady = getState(EngineState).userReady
```
Comment (HexaField) [inline packages/client-core/src/user/components/UserMenu/menus/AvatarSelectMenu.tsx:64]: This needs to be `useHookstate(getMutableState(EngineState).userReady)`

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/client-core/src/user/components/UserMenu/menus/AvatarSelectMenu.tsx (hunk 1 above)
Comment (aditya-mitra) [reply packages/client-core/src/user/components/UserMenu/menus/AvatarSelectMenu.tsx]: Implemented!

**Review by HexaField — APPROVED** [review]

**Pull request merged**